exchanges/gateio/gateio_wrapper.go

```
@@ -1262,62 +1278,60 @@ func (g *Gateio) CancelAllOrders(ctx context.Context, o *order.Cancel) (order.Ca
+			return cancelAllOrdersResponse, currency.ErrCurrencyPairEmpty
+		}
+		if o.Pair.IsEmpty() {
+			return cancelAllOrdersResponse, currency.ErrCurrencyPairEmpty
+		}
+		settle, err := g.getSettlementFromCurrency(o.Pair)
 		if err != nil {
 			return cancelAllOrdersResponse, err
 		}
-		for i := range contracts {
-			settle, err := g.getSettlementFromCurrency(contracts[i])
-			if err != nil {
-				continue
-			}
-			cancel, err := g.CancelMultipleFuturesOpenOrders(ctx, contracts[i], o.Side.Lower(), settle)
-			if err != nil && len(cancelAllOrdersResponse.Status) != 0 {
-				return cancelAllOrdersResponse, err
-			}
-			for f := range cancel {
-				cancelAllOrdersResponse.Status[strconv.FormatInt(cancel[f].ID, 10)] = cancel[f].Status
-			}
+		cancel, err := g.CancelMultipleFuturesOpenOrders(ctx, o.Pair, o.Side.Lower(), settle)
+		if err != nil && len(cancelAllOrdersResponse.Status) != 0 {
+			return cancelAllOrdersResponse, err
+		}
+		for f := range cancel {
+			cancelAllOrdersResponse.Status[strconv.FormatInt(cancel[f].ID, 10)] = cancel[f].Status
 		}
 	case asset.DeliveryFutures:
-		contracts, err := g.GetAvailablePairs(asset.DeliveryFutures)
+		if o.Pair.IsEmpty() {
+			return cancelAllOrdersResponse, currency.ErrCurrencyPairEmpty
+		}
+		settle, err := g.getSettlementFromCurrency(o.Pair)
 		if err != nil {
 			return cancelAllOrdersResponse, err
 		}
-		for i := range contracts {
-			settle, err := g.getSettlementFromCurrency(contracts[i])
-			if err != nil {
-				continue
-			}
-			cancel, err := g.CancelMultipleDeliveryOrders(ctx, contracts[i], o.Side.Lower(), settle)
-			if err != nil && len(cancelAllOrdersResponse.Status) != 0 {
-				return cancelAllOrdersResponse, err
-			}
-			for f := range cancel {
-				cancelAllOrdersResponse.Status[strconv.FormatInt(cancel[f].ID, 10)] = cancel[f].Status
-			}
+		cancel, err := g.CancelMultipleDeliveryOrders(ctx, o.Pair, o.Side.Lower(), settle)
+		if err != nil && len(cancelAllOrdersResponse.Status) != 0 {
```

So ignore errors if status is len 0?